src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -745,7 +745,17 @@ ExtractInsertPartitionValue(Query *query, Var *partitionColumn)
 												partitionColumn->varattno);
 	if (targetEntry != NULL)
 	{
-		Assert(IsA(targetEntry->expr, Const));
+		if (IsA(targetEntry->expr, Param))
+		{
+			ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+							errmsg("cannot execute a prepared insert statement on "
```

Comment:
> I wonder if we shouldn't just always error out if there's a param on the partition key instead. That'll be a lot easier for users to understand.

Mostly, users' drivers are running PREPARE, and generally, these drivers run the prepared statement just once. Also, we have unnamed prepared statements workaround. For reference #306. Erroring at the first trial will limit our coverage.

Given this, I would prefer to error out after 5th iteration. I will think about the error message, and I'm open to suggestions. What do you think?